huaweicloud/services/coc/resource_huaweicloud_coc_war_room.go

```diff
@@ -0,0 +1,440 @@
+package coc
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+
+	"github.com/chnsz/golangsdk"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+var warRoomNonUpdatableParams = []string{"war_room_name", "description", "region_code_list", "application_id_list",
+	"incident_number", "schedule_group", "schedule_group.*.role_id", "schedule_group.*.scene_id", "participant",
+	"war_room_admin", "application_names", "region_names", "enterprise_project_id", "notification_type"}
+
+// @API COC POST /v1/external/warrooms
+// @API COC POST /v1/external/warrooms/list
+func ResourceWarRoom() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceWarRoomCreate,
+		ReadContext:   resourceWarRoomRead,
+		UpdateContext: resourceWarRoomUpdate,
+		DeleteContext: resourceWarRoomDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		CustomizeDiff: config.FlexibleForceNew(warRoomNonUpdatableParams),
+
+		Schema: map[string]*schema.Schema{
+			"war_room_name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"application_id_list": {
+				Type:     schema.TypeList,
+				Required: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"incident_number": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"schedule_group": {
+				Type:     schema.TypeList,
```

Comment:
```suggestion
				Type:     schema.TypeSet,
```
and fix other places